src/backend/access/aocs/aocsam_handler.c

```diff
@@ -590,12 +590,14 @@ extractcolumns_from_node(Node *expr, bool *cols, AttrNumber natts)
 }
 
 static TableScanDesc
-aoco_beginscan_extractcolumns(Relation rel, Snapshot snapshot, ParallelTableScanDesc parallel_scan,
-							  List *targetlist, List *qual,
-							  uint32 flags)
+aoco_beginscan_extractcolumns(Relation rel, Snapshot snapshot, int nkeys, struct ScanKeyData *key,
+							  ParallelTableScanDesc parallel_scan,
+							  PlanState *ps, uint32 flags)
 {
 	AOCSScanDesc	aoscan;
 	AttrNumber		natts = RelationGetNumberOfAttributes(rel);
+	List *targetlist = ps->plan->targetlist;
+	List *qual = ps->plan->qual;
```

Comment:
key is not used by aoco?

Reply:
Normal table scan will have null scan key. It's used by runtime filter scan keys. AoCo has its own implementation on how to use these scan keys to filter tuples. The implementation is not yet ready now

Reply:
OK,  we need to pass bloom filter through it.